.github/workflows/test.yml

```
@@ -1,6 +1,10 @@
 name: Test
 
-on: [push, pull_request]
+on:
+  push:
+    branches:
+    - master
```

The [push event](https://github.com/open-telemetry/opentelemetry-python/actions?query=event%3Apush+workflow%3ATest) is only created if you push directly to the repo, instead of pushing a branch from your own.

Having the CI trigger on pushes to all branches has the advantage that you can automatically run the CI on your own repo, without having to work on master or creating PRs to yourself.

I actually make use of that, so I would suggest [ignoring](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#example-ignoring-branches-and-tags) release branches instead:
```suggestion
    branches-ignore:
    - 'release/*'
```

---

It may be worth noting that having the CI run on all branches is especially useful when testing changes to the CI pipeline itself.